src/goto-programs/interpreter.cpp

```diff
@@ -500,7 +500,8 @@ exprt interpretert::get_value(
     }
     return result;
   }
-  if(use_non_det && (memory[offset].initialized>=0))
+  if(use_non_det &&
+     memory[offset].initialized!=memory_cellt::initializedt::WRITTEN_BEFORE_READ)
```

Comment:
Yes - why would one return a non-det value in those cases, but _not_ when READ_BEFORE_WRITTEN?